files/shaders/objects_fragment.glsl

```diff
@@ -42,9 +42,11 @@ uniform sampler2D specularMap;
 varying vec2 specularMapUV;
 #endif
 
+#if !@accurateFog
 varying float depth;
+#endif
 
-#define PER_PIXEL_LIGHTING (@normalMap || @forcePPL)
+#define PER_PIXEL_LIGHTING (@normalMap || @forcePPL || @accurateFog)
```

Comment:
We certainly don't want to force per-pixel lighting when using your fog. Morrowind's assets are stupid and regularly look wrong if they're not done with vertex lighting, so we avoid PPL wherever possible. There's no reason why your fog would make it a requirement.